### PR TITLE
display percentage of achievement of todo in markdown

### PR DESCRIPTION
**Pull request by sosukesuzuki**
I heard that There are people who using Boostnote as todo list. So, I made percentage of achievement it visible.

todolist

### CODE REVIEW

**Comment by LetItRock** [comment]
Hello @sosukesuzuki,
I have a question - maybe it would be better to show something like this: "2 of 6 tasks complete" instead of percentage progress, like the same way as Github does?

**Comment by sosukesuzuki** [comment]
Thank you for commenting @LetItRock ! That might be nice! I will discuss it.

File: browser/main/Detail/MarkdownNoteDetail.js [hunk 1]
```diff
@@ -95,6 +96,24 @@ class MarkdownNoteDetail extends React.Component {
...
   }
 
+  getPercentageOfCompleteTodo (value) {
+    let splitted = value.split('\n')
+    let numberOfTodo = 0
+    let numberOfCompletedTodo = 0
+
+    for (let i = 0; i < splitted.length; i++) {
+      let trimmedLine = splitted[i].trim()
+      if (trimmedLine.match(/^- \[\s|x\] ./)) {
```
Comment (asmsuechan) [inline browser/main/Detail/MarkdownNoteDetail.js]: You need to add `+` and `*` as statements.

### +

screen shot 2017-05-06 at 08 20 40

### *
screen shot 2017-05-06 at 08 22 05

File: browser/main/Detail/MarkdownNoteDetail.js [hunk 2]
```diff
@@ -95,6 +96,24 @@ class MarkdownNoteDetail extends React.Component {
     return title
   }
 
+  getPercentageOfCompleteTodo (value) {
+    let splitted = value.split('\n')
+    let numberOfTodo = 0
+    let numberOfCompletedTodo = 0
+
+    for (let i = 0; i < splitted.length; i++) {
```
Comment (asmsuechan) [inline browser/main/Detail/MarkdownNoteDetail.js]: I assume `forEach` is better than `for` in this case.

File: browser/main/Detail/MarkdownNoteDetail.js [hunk 3]
```diff
@@ -95,6 +96,24 @@ class MarkdownNoteDetail extends React.Component {
     return title
   }
 
+  getPercentageOfCompleteTodo (value) {
```
Comment (asmsuechan) [inline browser/main/Detail/MarkdownNoteDetail.js]: I prefer the name of the argument is `noteContent`.

File: browser/components/TodolistPercentage.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+/**
+ * @param {number} percentageOfTodo
+ */
+
+const TodoListPercentage = ({
+  percentageOfTodo
+}) => (
+  <div styleName='percentageBar' style={{display: isNaN(percentageOfTodo) ? 'none' : ''}}>
+    <div styleName='progressBar' style={{width: percentageOfTodo + '%'}}>
+      <p styleName='percentageText'>{percentageOfTodo + '%'}</p>
```
Comment (asmsuechan) [inline browser/components/TodolistPercentage.js]: You can use string literal :smile:

File: browser/main/Detail/MarkdownNoteDetail.js (hunk 3 above)
Comment (sosukesuzuki) [inline browser/main/Detail/MarkdownNoteDetail.js]: Thank you for commenting ! Surely it is difficult to understand it mean. I'll fix it.

File: browser/main/Detail/MarkdownNoteDetail.js (hunk 2 above)
Comment (sosukesuzuki) [inline browser/main/Detail/MarkdownNoteDetail.js]: Thank you for commenting!  Surely, It's long to using 'for' than using 'foreach'. So, I'll fix it.

File: browser/components/TodolistPercentage.js (hunk 4 above)
Comment (sosukesuzuki) [inline browser/components/TodolistPercentage.js]: Thank you for telling me. Oh...  I had overlooked `+` and `*`. I'll add these operator.

File: browser/main/Detail/MarkdownNoteDetail.js (hunk 1 above)
Comment (sosukesuzuki) [inline browser/main/Detail/MarkdownNoteDetail.js]: Thank you for telling me. Oh... I had overlooked + and *. I'll add these operator.

**Comment by kazup01** [comment]
I will fix this UI at sometime soon.

**Pull request merged**